cli/datapipelines.go

```diff
@@ -0,0 +1,309 @@
+	}
+
+	mqlBinary := current.GetMqlBinary()
+	if args.MQL != "" || args.MqlFile != "" {
+		mqlBinary, err = parseMQL(args.MQL, args.MqlFile)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = client.datapipelinesClient.UpdateDataPipeline(context.Background(), &datapipelinespb.UpdateDataPipelineRequest{
+		Id:        args.ID,
+		Name:      name,
+		Schedule:  schedule,
+		MqlBinary: mqlBinary,
+	})
+	if err != nil {
+		return fmt.Errorf("error updating data pipeline: %w", err)
+	}
+
+	printf(c.App.Writer, "%s (id: %s) updated.", name, args.ID)
+	return nil
+}
+
+type datapipelineDeleteArgs struct {
+	ID string
+}
+
+// DatapipelineDeleteAction deletes a data pipeline.
+func DatapipelineDeleteAction(c *cli.Context, args datapipelineDeleteArgs) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.datapipelinesClient.DeleteDataPipeline(context.Background(), &datapipelinespb.DeleteDataPipelineRequest{
+		Id: args.ID,
+	})
+	if err != nil {
+		return fmt.Errorf("error deleting data pipeline: %w", err)
+	}
+
+	printf(c.App.Writer, "data pipeline (id: %s) deleted.", args.ID)
+	return nil
+}
+
+type datapipelineDescribeArgs struct {
+	ID string
+}
+
+// DatapipelineDescribeAction describes a data pipeline and its status.
+func DatapipelineDescribeAction(c *cli.Context, args datapipelineDescribeArgs) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.datapipelinesClient.GetDataPipeline(context.Background(), &datapipelinespb.GetDataPipelineRequest{
+		Id: args.ID,
+	})
+	if err != nil {
+		return fmt.Errorf("error getting data pipeline: %w", err)
+	}
+	pipeline := resp.GetDataPipeline()
+
+	runsResp, err := client.datapipelinesClient.ListPipelineRuns(context.Background(), &datapipelinespb.ListPipelineRunsRequest{
+		Id:       args.ID,
+		PageSize: 1,
+	})
+	if err != nil {
+		return fmt.Errorf("error listing pipeline runs: %w", err)
+	}
+	runs := runsResp.Runs
+
+	mql, err := mqlJSON(pipeline.GetMqlBinary())
+	if err != nil {
+		warningf(c.App.Writer, "error parsing MQL query: %s", err)
+		mql = "(error parsing MQL query)"
+	}
+
+	printf(c.App.Writer, "ID: %s", pipeline.GetId())
+	printf(c.App.Writer, "Name: %s", pipeline.GetName())
+	printf(c.App.Writer, "Enabled: %t", pipeline.GetEnabled())
+	printf(c.App.Writer, "Schedule: %s", pipeline.GetSchedule())
+	printf(c.App.Writer, "MQL query: %s", mql)
+
+	var pipelineRunStatusMap = map[datapipelinespb.PipelineRunStatus]string{
+		datapipelinespb.PipelineRunStatus_PIPELINE_RUN_STATUS_UNSPECIFIED: "Unknown",
+		datapipelinespb.PipelineRunStatus_PIPELINE_RUN_STATUS_SCHEDULED:   "Scheduled",
+		datapipelinespb.PipelineRunStatus_PIPELINE_RUN_STATUS_STARTED:     "Running",
+		datapipelinespb.PipelineRunStatus_PIPELINE_RUN_STATUS_COMPLETED:   "Success",
+		datapipelinespb.PipelineRunStatus_PIPELINE_RUN_STATUS_FAILED:      "Failed",
+	}
+
+	if len(runs) > 0 {
+		printf(c.App.Writer, "Last run: %s, %s.",
```

Comment:
Nit: Doesn't provide any details around what time range the pipeline was run against

Reply:
added data range as well as end time printing